src/govuk/components/notification-banner/template.test.js

```diff
@@ -0,0 +1,327 @@
+
+      const $component = $('.govuk-notification-banner')
+      expect($component.hasClass('app-my-class')).toBeTruthy()
+    })
+
+    it('renders attributes', () => {
+      const $ = render('notification-banner', examples.attributes)
+
+      const $component = $('.govuk-notification-banner')
+      expect($component.attr('my-attribute')).toEqual('value')
+    })
+  })
+
+  describe('html', () => {
+    it('renders title as escaped html when passed as text', () => {
+      const $ = render('notification-banner', examples['title html as text'])
+      const $title = $('.govuk-notification-banner__title')
+
+      expect($title.html().trim()).toEqual('&lt;span&gt;Important information&lt;/span&gt;')
+    })
+
+    it('renders title as html', () => {
+      const $ = render('notification-banner', examples['title as html'])
+      const $title = $('.govuk-notification-banner__title')
+
+      expect($title.html().trim()).toEqual('<span>Important information</span>')
+    })
+
+    it('renders content as escaped html when passed as text', () => {
+      const $ = render('notification-banner', examples['html as text'])
+      const $content = $('.govuk-notification-banner__content')
+
+      expect($content.html().trim()).toEqual('&lt;span&gt;This publication was withdrawn on 7 March 2014.&lt;/span&gt;')
+    })
+
+    it('renders content as html', () => {
+      const $ = render('notification-banner', examples['with text as html'])
+      const $contentHtml = $('.govuk-notification-banner__content')
+
+      expect($contentHtml.html().trim()).toEqual('<h3 class="govuk-heading-m">This publication was withdrawn on 7 March 2014</h3><p>Archived and replaced by the <a href="#">new planning guidance</a> launched 6 March 2014 on an external website</p>')
+    })
+  })
+
+  describe('when success type is passed', () => {
+    it('renders with appropriate class', () => {
+      const $ = render('notification-banner', examples['with type as success'])
+
+      const $component = $('.govuk-notification-banner')
+      expect($component.hasClass('govuk-notification-banner--success')).toBeTruthy()
+    })
+
+    it('has role=alert attribute', () => {
+      const $ = render('notification-banner', examples['with type as success'])
+
+      const $component = $('.govuk-notification-banner')
+      expect($component.attr('role')).toEqual('alert')
+    })
+
+    it('does not render aria-labelledby', () => {
+      const $ = render('notification-banner', examples['with type as success'])
+      const $component = $('.govuk-notification-banner')
+
+      expect($component.attr('aria-labelledby')).toBeFalsy()
+    })
+
+    it('does not render a title id for aria-labelledby', () => {
+      const $ = render('notification-banner', examples['with type as success'])
+      const $component = $('.govuk-notification-banner')
+
+      expect($component.attr('id')).not.toEqual('govuk-notification-banner-title')
```

Comment:
Should this be checking for undefined, as we don't expect an id at all now?

Reply:
Done!